### ISSUE TITLE
Dail TCP Lookup Connection Error

### ISSUE THREAD

**Issue by mskian** [termux/termux-app#1138]
I am running a CLI based on Golang
While trying to connect any URLs I got this Below Error 

```
Error: Get https://google.com/version: dial tcp: lookup google.com on [::1]:53: read udp [::1]:48773->[::1]:53: read: connection refused
```

**Comment by xeffyr** [comment]
Is your program compiled with `GOOS=android` ?

**Comment by mskian** [comment]
DNS Resolver is `8.8.8.8`
CLI I am Installing  - https://github.com/gotify/cli

**Comment by xeffyr** [comment]
> DNS Resolver is 8.8.8.8

This not a DNS resolver !
DNS resolver is part of code that handles DNS queries. Typically handled via CGO calls to C library: GNU libc on Linux and Bionic on Android. They are incompatible.

Note that static linking of the executable does not have any effect. Resolver from GNU libc always used via dlopen().

**Comment by mskian** [comment]
@xeffyr Any Solution for this?

**Comment by xeffyr** [comment]
If CGO was disabled at build time, binary will look for file /etc/resolv.conf. In such case you should be able to run binary with `termux-chroot` from package `proot`.

**Comment by mskian** [comment]
@xeffyr Example Please

**Comment by xeffyr** [comment]
Perhaps it may be easier to recompile it on device. Do:
```
pkg install golang
go get github.com/gotify/cli
```

Binary "cli" will be available at `~/go/bin`.

Other workarounds are not exist: either use binary in environment it was compiled for or recompile binary for Termux.

**Comment by mskian** [comment]
`termux-chroot`
Connects the URL Successfully

but got this below error (Using Binary file `arm64`)

```
user: Current not Implemented on linux/arm64
```

**Comment by xeffyr** [comment]
> Current not Implemented on linux/arm64

Try arm64 binary compiled specially for Termux: [gotify-cli.gz](https://github.com/termux/termux-app/files/3199528/gotify-cli.gz).

Or use method shown above to compile it yourself.

**Comment by xeffyr** [comment]
> Current not Implemented on linux/arm64

Try to use a binary compiled specially for Termux/android: [gotify-cli.gz](https://github.com/termux/termux-app/files/3199590/gotify-cli.gz) (arm64).

**Comment by mskian** [comment]
Thanks a lot, it's working 
How you build this File please share the Steps Here

**Comment by mskian** [comment]
Share the Steps to Build binary compiled Package for Termux

**Comment by mskian** [comment]
> Perhaps it may be easier to recompile it on device. Do:
> 
> ```
> pkg install golang
> go get github.com/gotify/cli
> ```
> 
> Binary "cli" will be available at `~/go/bin`.
> 
> Other workarounds are not exist: either use binary in environment it was compiled for or recompile binary for Termux.

Follow this step working

**Comment by mvolfik** [comment]
Hello, I'm trying to compile netlify/netlify-credential-helper, which doesn't provide a binary for Android. By modifying makefile I managed to build for `os=android arch=arm64` and re-enable CG0, so I stopped getting errors on resolve, but now I'm getting the error `&{0 } (*models.Error) is not supported by the TextConsumer, can be resolved by supporting TextUnmarshaler interface`. I'm not a Go programmer - anyone has an idea what this could mean and if it's fixable? Should I create a new issue (and if so, here or in their repo?)? Thanks a lot!
